OTA-2653 Move "Getting Started Page

merltron opened the pull request:
We have to move the getting started page from the web app to the docs site. I thought the intro page would be a good place for it since this is true "end-to-end" getting started content.

The web app is running out of space so we can no longer host a dedicated "Getting Started" page there. See the parent case for more details on the proposed UI changes.

merltron committed:
OTA-2653 Move "Getting Started Page

We have to move the getting started page
from the web app to the docs site. I
thought the intro page would be a good
place for it since this is true
"end-to-end" getting started content.

Signed-off-by: merltron <30755179+merltron@users.noreply.github.com>

tkfu commented on _posts/2018-02-08-start-intro.adoc:
```diff
@@ -4,7 +4,267 @@
+
+*To download a provisioning key, follow these steps:*
+
+1.  Open the OTA Connect Portal, open the menu for your user profile and click *Provisioning Keys*
```
This should be a link.

tkfu commented on _posts/2018-02-08-start-intro.adoc:
```diff
@@ -4,7 +4,267 @@
+
+You‘ll need to copy the zip file to the file system of your device.
+
+image:../images/s1-prov.svg[image]
```
This is a general comment for all the images: why are they SVG files with low-res PNG data embedded inside in base64?

tkfu commented on _posts/2018-02-08-start-intro.adoc:
```diff
@@ -4,7 +4,267 @@
+
+Once you have provisioned your devices, you should see them show up in the OTA Connect user interface.
+
+image:../images/s2-find.svg[image]
```
This screenshot is from the demo account, and has fake dashboard features. For the docs, we should be using the real, current UI.

merltron replied:
@tkfu  Sorry, I don't follow - what are the fake features you see? I just compared this page in "real" and "demo" and they look pretty much the same (apart from the devices listed).

tkfu replied:
The bottom section with the charts. They show features (like cert monitoring and forecasting connection numbers) that aren't yet available.

merltron replied:
ahh.  I realize the confusion now. I was using Benes account which I thought was "normal" and comparing it to the "demo" account which is "demo". In reality, both accounts have demo features enabled. I created a login for myself and the alpha features are gone.

merltron replied:
fixed

tkfu commented on _posts/2018-02-08-start-intro.adoc:
```diff
@@ -4,7 +4,267 @@
+
+*To check if your devices have been provisioned properly, follow this step:*
+
+* Navigate to the *Devices* page and select the *All Devices* or *Ungrouped Devices* menu items.
```
Again, can be linked directly.

tkfu commented on _posts/2018-02-08-start-intro.adoc:
```diff
@@ -4,7 +4,267 @@
+
+*To create a smart group, follow these steps:*
+
+1.  Open the *Devices* tab and click *+Add group*.
```
Devices tab can be linked.

tkfu commented on _posts/2018-02-08-start-intro.adoc:
```diff
@@ -4,7 +4,267 @@
+
+To install the first version of your software on a test device, follow these steps:
+
+1.  Open the device details that we first looked at in === Step 3
```
Should be an anchor link. You can do it by linking to a slugified version of the section title, but best practice is to actually assign an ID, and then use either the bare xref syntax or the custom link text:

```
[#group-devices]
=== Step 3: Group Your Devices

You can follow <<group-devices>> to <<group-devices,learn to add devices to groups>>.
```

tkfu commented on _posts/2018-02-08-start-intro.adoc:
```diff
@@ -4,7 +4,267 @@
+To install the first version of your software on a test device, follow these steps:
+
+1.  Open the device details that we first looked at in === Step 3
+2.  Navigate to **Devices**, select a device group, and select a test device within that group.
```
Devices tab can be directly linked.

tkfu commented on _posts/2018-02-08-start-intro.adoc:
```diff
@@ -4,7 +4,267 @@
+
+=== Step 6: Create a Software Update Unit
+
+When you create a software update, you need to define two basic „assignment criteria“.
```
Don't use German smart quotes.

tkfu commented on _posts/2018-02-08-start-intro.adoc:
```diff
@@ -4,7 +4,267 @@
+
+You might be wondering how to select the vehicle fleet that is supposed to receive this update.
+
+That step is covered when you create a *Campaign* which we‘ll get to next...
```
Missing a comma.

tkfu commented on _posts/2018-02-08-start-intro.adoc:
```diff
@@ -4,7 +4,267 @@
+1.  Click *Campaigns* and click *Create Campaign*.
+2.  In the wizard that appears, enter a campaign name and click *Next*.
+3.  Select the device groups that you want to deploy the update to.
+* In the second step of our Getting Started guide, we showed you how to create a smart group.
```
I believe this was the 3rd step. This is why it's good to use the x-ref syntax.

tkfu commented on _posts/2018-02-08-start-intro.adoc:
```diff
@@ -4,7 +4,267 @@
+
+However, since this is a Get Started guide, we‘re going to describe the easy way.
+
+1.  To upload some software, follow these steps:
```
Feels wrong for step 1 of a process to be "follow these steps". This sentence should probably be outside of the ordered list.

tkfu commented on _posts/2018-02-08-start-intro.adoc:
```diff
@@ -4,7 +4,267 @@
+* The *hard way* is to locally build a filesystem image that is OTA-enabled and already has your software installed.
++
+During the build process, the filesystem image (with your new software) is automatically uploaded to the software repository on your OTA Connect server. This method is closer to how software gets installed in production.
+** For more information about this method, see the https://docs.ota.here.com/quickstarts/start-intro.html[developer documentation].
```
Remove one of the asterisks--it's not rendering properly right now.

merltron replied:
fixed

tkfu commented on _posts/2018-02-08-start-intro.adoc:
```diff
@@ -4,7 +4,267 @@
+4.  Click the *first version* that you uploaded.
+* In the right-hand *PROPERTIES* panel, you‘ll see more details about the selected software version — the status should be *Not Installed*
++
+.
```
Looks like you added the newline before the period instead of after.

merltron replied:
fixed

tkfu commented on _posts/2018-02-08-start-intro.adoc:
```diff
@@ -4,7 +4,267 @@
+* The next few steps are optional for now, so we‘re going to leave them at their default settings. Here‘s a brief explanation of those steps:
+** *Distribution settings* – Configure the campaign so that end users must consent to the update. You can define your own notification text for the end user to read.
+** *Dependencies Management* – Configure the campaign to check for required software dependencies before installing the update.
+** *Programming Sequencer* – Configure the sequence in which each piece of software is installed. You can also configure some rollback behavior in case any piece of software failed to install.
```
Again, we shouldn't put demo features that aren't actually available in the docs.

merltron replied:
fixed

tkfu commented on _posts/2018-02-08-start-intro.adoc:
```diff
@@ -4,7 +4,267 @@
+Click *Campaigns* and click a status tab.
+
+* Assuming your campaign is still running you would click the *Running* tab.
+* If your campaign is a test campaign, it might finish quickly, in which case, you‘ll find it on the *Finished* tab
```
Missing period

merltron replied:
fixed

tkfu commented on _posts/2018-02-08-start-intro.adoc:
```diff
@@ -4,7 +4,267 @@
+2.  Search for a device that you know is running the defective software and open the device details for that device
+3.  In the HARDWARE section, locate the primary ECU and click the info icon image:../images/info.svg[Icon].
+4.  In the window that appears, click the *Packages* tab and use the search box to filter for the defective software version.
+5.  Click the blacklist icon image:../images/ban_grey.png[Icon] next to the version number.
```
Image renders gigantic--please set height/width attributes.

tkfu reviewed (changes requested):
I think it's important to replace the screenshots with ones that don't show features that aren't available to the general public. Because they'll need to be redone for that purpose anyway, let's also fix the images so that they aren't PNG embedded in SVG.

Commit:
OTA-2653 Clean up issues on intro/getstarted page

Fixed formatting issues
Added linking suggestions
Added "thumb" role so that screenshots get borders
Removed Alpha junk.

Signed-off-by: Jon Oster <jon.oster@here.com>

merltron commented:
@tkfu OK I think I fixed everything - have another look and let me know if it's OK:

tkfu committed:
minor typo/formatting fixes

Signed-off-by: Jon Oster <jon.oster@here.com>

tkfu reviewed (approved):
I added a minor fix-up commit (13bb3a1), and squashed the image editing commits. I'm happy with this as-is; might be a good idea to double-check 13bb3a1 before merging.

merltron commented:
Thanks - 13bb3a1 looks fine.
